### ISSUE TITLE
Push not working in production build

### ISSUE THREAD

**Issue by puskaiz** [vaadin/hilla#786]
Using Hilla with Spring boot starter (and LIT) push feature is not working in Production mode.

Hilla version 1.3.5
Spring boot version 2.7.8
Node version 18.13.0
Java version 17
(Embedded Tomcat)

**Comment by Artur-** [comment]
Seems to work fine for me

**Comment by puskaiz** [comment]
I have reproduced today with flux-clock (this time with react, not lit):
followed  the tutorial from here: 
https://hilla.dev/docs/react/guides/reactive-endpoints#example

mvn clean package -Pproduction

java -jar target/flux-clock-1.0-SNAPSHOT.jar -Dvaadin.hillaPush=true 

I see the output:
The following EXPERIMENTAL features are enabled:
- Push support in Hilla

I see an exception in the logs:
 No AtmosphereHandler found. Make sure you define it inside WEB-INF/atmosphere.xml or annotate using @___Service

And the clock is not working

From Intellij it works.

**Comment by cromoteca** [comment]
I was able to reproduce it on Hilla 1.3.6. Version 2.0.0.beta5 works fine instead.

**Comment by puskaiz** [comment]
will the fix be released in ga?

**Comment by cproinger** [comment]
we have experienced the same Problem ("No AtmosphereHandler found"). Can you suggest a workaround? we cannot upgrade to hilla v2 at the moment

**Comment by cproinger** [comment]
i tried to implemented a workaround by adding `dev.hilla.push.PushConfigurer` to `spring.autoconfigure.exclude` and replacing it with the following config-class (code from https://github.com/vaadin/hilla/commit/ede47cd1887528e96386b7ddc5472ffdf9c6d79a). 
```
@Configuration
@ConditionalOnFeatureFlag("hillaPush")
public class HillaServerPushConfig {

    public static final String HILLA_PUSH_PATH = "/HILLA/push";

    @Bean
    public PushEndpoint pushEndpoint() {
        return new PushEndpoint();
    }

    @Bean
    public PushMessageHandler pushMessageHandler(
            final EndpointInvoker endpointInvoker) {
        return new PushMessageHandler(endpointInvoker);
    }

    @Bean
    public EmbeddedAtmosphereInitializer atmosphereInitializer() {
        return new EmbeddedAtmosphereInitializer();
    }

    @Bean
    @Primary
    public ServletRegistrationBean<AtmosphereServlet> atmosphereServlet(final PushEndpoint pushEndpoint) {
        final AtmosphereServlet atmosphereServlet = new AtmosphereServlet();
        final ServletRegistrationBean<AtmosphereServlet> registration = new ServletRegistrationBean<>(
                atmosphereServlet, HILLA_PUSH_PATH);

        final List<AtmosphereInterceptor> interceptors = Arrays.asList(
                new AtmosphereResourceLifecycleInterceptor(),
                new TrackMessageSizeInterceptor(),
                new SuspendTrackerInterceptor());
        final AtmosphereFramework fw = atmosphereServlet.framework();
        fw.setDefaultBroadcasterClassName(SimpleBroadcaster.class.getName());
        fw.addAtmosphereHandler(HILLA_PUSH_PATH, pushEndpoint, interceptors);

        // Override the global mapping set by Flow
        registration.addInitParameter(ApplicationConfig.JSR356_MAPPING_PATH,
                HILLA_PUSH_PATH);
        registration.setLoadOnStartup(0);
        registration.setOrder(Ordered.HIGHEST_PRECEDENCE);
        return registration;
    }

    private static class EmbeddedAtmosphereInitializer
            extends ContainerInitializer implements ServletContextInitializer {

        @Override
        public void onStartup(final ServletContext servletContext)
                throws ServletException {
            onStartup(Collections.<Class<?>>emptySet(), servletContext);
        }

    }
}
```
This does work if my EndpointSubscription Endpoint-Method does not require an Authorization. If it does, i get  an access-denied error on the client. 

```
@Endpoint
public class ServerPushEndpoint {

    private static final Logger LOGGER = LoggerFactory.getLogger(ServerPushEndpoint.class);

    private final NotificationSinkEventListener notificationSubscriptionService;

    public ServerPushEndpoint(final NotificationSinkEventListener notificationSubscriptionService) {
        this.notificationSubscriptionService = notificationSubscriptionService;
    }

    @RolesAllowed("Reader")
    public EndpointSubscription<@Nonnull String> getNotifications() {
        return EndpointSubscription.of(notificationSubscriptionService.getNotificationEventFlux()
                .map(IJsonService.SIMPLE_JSON::toJson), () -> LOGGER.info("Notification-Subscription has been cancelled"));
    }
}
```

The code worked in 1.2.10 so maybe session-information is no longer propagated correctly to the web-socket-infrastructure or something like that?

**Comment by taefi** [comment]
This is fixed now and can be evaluated by using the 1.3-SNAPSHOT. 
This will be released with the next maintenance release (1.3.19) shortly.

If the problem still persists, please feel free to reopen this issue, or create another issue if the problem is not exactly the same.
